core/backend-itwin-client/src/oidc/AgentAuthorizationClient.ts

```diff
@@ -58,39 +58,19 @@ export class AgentAuthorizationClient extends BackendAuthorizationClient impleme
       throw new BentleyError(AuthStatus.Error, error.message || "Authorization error", Logger.logError, loggerCategory, () => ({ error: error.error, message: error.message }));
     }
 
-    const userProfile = tokenSet.access_token
-      ? decode(tokenSet.access_token, { json: true, complete: false })
-      : undefined;
-    this._accessToken = AccessToken.fromTokenResponseJson(tokenSet, userProfile);
+    this._accessToken = tokenSet.access_token;
+    this._expiresAt = new Date(tokenSet.expires_at!); // TODO: Check if this is in proper format

```

Comment:
we should NOT be using the non null assertion

Reply:
Should be fixed now.